sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/schema/primitive/AvroLongSchema.java

```diff
@@ -0,0 +1,92 @@
+package com.azure.storage.internal.avro.implementation.schema.primitive;
+
+import com.azure.storage.internal.avro.implementation.AvroParserState;
+import com.azure.storage.internal.avro.implementation.schema.AvroSchema;
+
+import java.util.function.Consumer;
+
+/**
+ * long values are written using variable-length zig-zag coding.
+ *
+ * The equivalent while loop version of the code would look like this.
+ *
+ * byte b = this.state.consume() & 0xFF;
+ * long n = b & 0x7F;
+ * long shift = 7;
+ * while ((b & 0x80) != 0) {
+ *  b = this.state.consume() & 0xFF;
+ *  next |= (b & 0x7F) << shift;
+ *  shift += 7;
+ * }
+ * long value = next;
+ * return (n >> 1) ^ -(n & 1);
+ */
+public class AvroLongSchema extends AvroSchema<Long> {
+
+    private long n; /* Keeps track of the number so far. */
+    private long shift = 7; /* The current shift value. */
+    private boolean first = true; /* Whether or not the first byte has been read yet. (This is to deal with the initial
+    code that runs before the while loop in the class level docs.)*/
+    private int lastB; /* The last byte read. */
+
+    /**
+     * Constructs a new AvroLongSchema.
+     *
+     * @param state The state of the parser.
+     * @param onResult The result handler.
+     */
+    public AvroLongSchema(AvroParserState state, Consumer<Long> onResult){
+        super(state, onResult);
+    }
+
+    @Override
+    public void add() {
+        this.state.push(this);
+    }
+
+    /**
+     * Consuming bytes written into AvroParserState
```

Comment:
Might we worth describing little bit more about "variable lenght zig-zag encoding" per Avro official doc (or reference there from here)

Reply:
I'll add a doc.